test/cql-pytest/test_permissions.py

```diff
@@ -553,3 +554,174 @@ def test_native_functions_always_exeutable(cql):
+            schema = "p int primary key"
+            table = f'{test_keyspace}.{unique_name()}'
+            with new_named_table(user1_session, table, schema):
+                # A minimal check that auto-grant actually worked for user1,
+                # user2 can't read the table.
+                user1_session.execute(f'SELECT * FROM {table}')
+                with pytest.raises(Unauthorized):
+                    user2_session.execute(f'SELECT * FROM {table}')
+            # At this point, table was deleted by user1, so user2 can create
+            # a table with the same name "table".
+            with new_named_table(user2_session, table, schema):
+                # Now, permissions were auto-granted for user2, but user1
+                # no longer has permissions on this table.
+                eventually_authorized(lambda: user2_session.execute(f'SELECT * FROM {table}'))
+                eventually_unauthorized(lambda: user1_session.execute(f'SELECT * FROM {table}'))
+
+def test_auto_revoke_view(cql, test_keyspace):
+    with new_user(cql) as user1, new_user(cql) as user2:
+        # Grant user1 and user2 CREATE permissions on test_keyspace, but no
+        # other permissions:
+        grant(cql, 'CREATE', 'KEYSPACE ' + test_keyspace, user1)
+        grant(cql, 'CREATE', 'KEYSPACE ' + test_keyspace, user2)
+        with new_session(cql, user1) as user1_session, new_session(cql, user2) as user2_session:
+            schema = "p int primary key, v int"
+            mv = f'{test_keyspace}.{unique_name()}'
+            with new_test_table(user1_session, test_keyspace, schema) as table:
+                # Create a view with a pre-chosen name mv as user1:
+                user1_session.execute(f'CREATE MATERIALIZED VIEW {mv} AS SELECT * FROM {table} WHERE p IS NOT NULL AND v IS NOT NULL PRIMARY KEY (v, p)')
+                # user1 was auto-granted permissions to read this view, user2
+                # was not:
+                user1_session.execute(f'SELECT * FROM {mv}')
+                with pytest.raises(Unauthorized):
+                    user2_session.execute(f'SELECT * FROM {mv}')
+                user1_session.execute(f'DROP MATERIALIZED VIEW {mv}')
+                # After user1 dropped the view with name mv, we let user2
+                # create a different table but reuse the same view name mv.
+                # We want to check that user1 no longer has permissions on
+                # this view:
+                with new_test_table(user2_session, test_keyspace, schema) as table2:
+                    user2_session.execute(f'CREATE MATERIALIZED VIEW {mv} AS SELECT * FROM {table2} WHERE p IS NOT NULL AND v IS NOT NULL PRIMARY KEY (v, p)')
+                    eventually_authorized(lambda: user2_session.execute(f'SELECT * FROM {mv}'))
+                    eventually_unauthorized(lambda: user1_session.execute(f'SELECT * FROM {mv}'))
+                    user2_session.execute(f'DROP MATERIALIZED VIEW {mv}')
```

Comment:
Nit: this can probably be de-dented once, right? Was it intentional that this is executed within the scope of `new_test_table(user1_session, test_keyspace, schema)`?

Reply:
Yes, it is intentional and necessary - CQL forbids deleting a table that has materialized views, so we need to execute DROP MATERIALIZED VIEW before we can go out of the scope and the table gets deleted.